ambry-clustermap/src/main/java/com.github.ambry.clustermap/CompositeClusterAgentsFactory.java

```diff
@@ -0,0 +1,91 @@
+
+
+/**
+ * A factory that creates a {@link CompositeClusterManager} and {@link ClusterParticipant}. Only one instance of each
+ * type of objects will ever be created by this factory.
+ */
+public class CompositeClusterAgentsFactory implements ClusterAgentsFactory {
+  private static final Logger logger = LoggerFactory.getLogger(CompositeClusterAgentsFactory.class);
+  private final StaticClusterAgentsFactory staticClusterAgentsFactory;
+  private final HelixClusterAgentsFactory helixClusterAgentsFactory;
+  private CompositeClusterManager compositeClusterManager;
+  private ClusterParticipant clusterParticipant;
+
+  /**
+   * Create an instance of this class.
+   * @param clusterMapConfig the {@link ClusterMapConfig} to use.
+   * @param hardwareLayoutFilePath the path to the hardware layout file.
+   * @param partitionLayoutFilePath the path to the partition layout file.
+   * @throws JSONException if there is an exception parsing the layout files.
+   * @throws IOException if there is an IO error accessing or reading the layout files.
+   */
+  public CompositeClusterAgentsFactory(ClusterMapConfig clusterMapConfig, String hardwareLayoutFilePath,
+      String partitionLayoutFilePath) throws JSONException, IOException {
+    staticClusterAgentsFactory =
+        new StaticClusterAgentsFactory(clusterMapConfig, hardwareLayoutFilePath, partitionLayoutFilePath);
+    helixClusterAgentsFactory = new HelixClusterAgentsFactory(clusterMapConfig, null, null);
+  }
+
+  /**
+   * Create an instance of this class.
+   * @param clusterMapConfig the {@link ClusterMapConfig} to use.
+   * @param partitionLayout the {@link PartitionLayout} to use with the internal {@link StaticClusterManager}
+   * @throws JSONException if there is an exception parsing the layout files.
+   * @throws IOException if there is an IO error accessing or reading the layout files.
+   */
+  CompositeClusterAgentsFactory(ClusterMapConfig clusterMapConfig, PartitionLayout partitionLayout)
+      throws JSONException, IOException {
+    staticClusterAgentsFactory = new StaticClusterAgentsFactory(clusterMapConfig, partitionLayout);
+    helixClusterAgentsFactory = new HelixClusterAgentsFactory(clusterMapConfig, null, null);
+  }
+
+  /**
+   * Create and return a {@link CompositeClusterManager}.
+   * @return the constructed {@link CompositeClusterManager}.
+   * @throws Exception if constructing the underlying {@link StaticClusterManager} or the {@link HelixClusterManager}
+   * throws an Exception.
+   */
+  @Override
+  public CompositeClusterManager getClusterMap() throws IOException {
+    if (compositeClusterManager == null) {
+      StaticClusterManager staticClusterManager = staticClusterAgentsFactory.getClusterMap();
+      HelixClusterManager helixClusterManager = null;
+      try {
+        helixClusterManager = helixClusterAgentsFactory.getClusterMap();
+      } catch (Exception e) {
+        logger.error("Helix cluster manager instantiation failed with exception", e);
```

Comment:
@vgkholla It seems there is a path in `CompositeClusterManager` to allow `null` for `HelixClusterManager`.